### PR TITLE
tests, utils: Move GetVMPodName to libpod

### PR DESCRIPTION
**Pull request by ormergi**
### What this PR does
Before this PR:
The helper `GetVMPodName` lives at test/utils.

After this PR:
`GetVMPodName` moved to libpod.

The `GetVMPodName` function is also refactored to address the following issues:
- Return error instead of assert the error
- Fetch virt client using the tests framework singleton

Fixes #

### Why we need it and why it was done in this way
The following tradeoffs were made:

The following alternatives were considered:

Links to places where the discussion took place: 

### Checklist

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

### Release note

```release-note
NONE
```

### CODE REVIEW

**Review by ShellyKa13 — COMMENTED** [review]
/lgtm
Nice divide of commits!

File: tests/libpod/query.go [hunk 1]
```diff
@@ -89,3 +89,27 @@ func vmiFieldSelector(vmi *v1.VirtualMachineInstance) string {
 	}
 	return strings.Join(fieldSelectors, ",")
 }
+
+func GetVmPodName(vmi *v1.VirtualMachineInstance) (string, error) {
```
[the remark this answers is not in the document]
Comment (ormergi) [reply tests/libpod/query.go:93]: Good catch, I posted a PR to make use of `GetPodByVirtualMachineInstance` instead, let see CI results.

**Review by enp0s3 — COMMENTED** [review]
@ormergi Can you please rebase?

File: tests/libpod/query.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (ormergi) [reply tests/libpod/query.go:93]: https://github.com/kubevirt/kubevirt/pull/12295

**Comment by ormergi** [comment]
Following https://github.com/kubevirt/kubevirt/pull/12256#discussion_r1661965499, I am closing this PR and continue at #12295, plesase have a look :) 
@ShellyKa13 @xpivarc @enp0s3 

/close

**Pull request closed without merging**